### PR TITLE
chapter 9 "Star/Quasar Classification ROC Curves" example trains classifiers on the whole data set rather than the train split

### PR DESCRIPTION
**Pull request by StKyr**
In fig_star_quasar_ROC.py, inside compute_results(), classifiers are trained on X rather than X_train. This means that the test set has been observed from the classifiers during training which of course is a bad practice.

The way to fix this would be to change line 90 from

model.fit(X, y)

to

model.fit(X_train, y_train)

Additionally, the figure fig_star_quasar_ROC_1.png needs to be updated as it is the result of the execution of the script.

### CODE REVIEW

**Comment by bsipocz** [comment]
The baseline figures are a totally separate issue, and they are anyway are the version from the first edition. Please remove changing it from this commit by amending it (I can come back a bit later and do it before merging if you prefer that way).

**Comment by StKyr** [comment]
> The baseline figures are a totally separate issue, and they are anyway are the version from the first edition. Please remove changing it from this commit by amending it (I can come back a bit later and do it before merging if you prefer that way).

Replaced the figure with the original one (from the master branch of the repo)

**Comment by bsipocz** [comment]
Thanks  @StKyr!
